graylog2-web-interface/src/components/event-definitions/event-definitions/SchedulingCell.tsx

```diff
@@ -0,0 +1,158 @@
+
+import { OverlayTrigger, Icon, Timestamp } from 'components/common';
+import { Popover, Button } from 'components/bootstrap';
+import { EventDefinitionsActions } from 'stores/event-definitions/EventDefinitionsStore';
+
+import type { Scheduler, EventDefinition } from '../event-definitions-types';
+
+type Props = {
+  definition: EventDefinition,
+};
+
+const WidePopover = styled(Popover)`
+  max-width: 500px;
+`;
+const DetailsList = styled.dl`
+`;
+
+const DetailTitle = styled.dt`
+  float: left;
+  clear: left;
+`;
+
+const DetailValue = styled.dd(({ theme }) => css`
+  margin-left: 180px;
+  word-wrap: break-word;
+
+  &:not(:last-child) {
+    border-bottom: 1px solid ${theme.colors.variant.lightest.default};
+    margin-bottom: 5px;
+    padding-bottom: 5px;
+  }
+`);
+
+const DetailsButton = styled(Button)`
+  padding: 6px 8px;
+`;
+
+const getTimeRange = (scheduler: Scheduler) => {
+  const from = scheduler?.data?.timerange_from;
+  const to = scheduler?.data?.timerange_to;
+
+  return (
+    <>
+      <DetailTitle>Next timerange:</DetailTitle>
+      <DetailValue><Timestamp dateTime={from} /> <Icon name="arrow-circle-right" /> <Timestamp dateTime={to} /></DetailValue>
+    </>
+  );
+};
+
+const detailsPopover = (title, scheduler: Scheduler, clearNotifications: () => void) => (
+  <WidePopover id="event-definition-details" title={`${title} details.`}>
+    <DetailsList>
+      <DetailTitle>Status:</DetailTitle>
+      <DetailValue>{scheduler.status}</DetailValue>
+      {scheduler.triggered_at && (
+      <>
+        <DetailTitle>Last execution:</DetailTitle>
+        <DetailValue><Timestamp dateTime={scheduler.triggered_at} /></DetailValue>
+      </>
+      )}
+      {scheduler.next_time && (
+      <>
+        <DetailTitle>Next execution:</DetailTitle>
+        <DetailValue><Timestamp dateTime={scheduler.next_time} /></DetailValue>
+      </>
+      )}
+      {getTimeRange(scheduler)}
+      <DetailTitle>Queued notifications:</DetailTitle>
+      <DetailValue>{scheduler.queued_notifications}
+        {scheduler.queued_notifications > 0 && (
+        <Button bsStyle="link" bsSize="xsmall" onClick={clearNotifications()}>
+          clear
+        </Button>
+        )}
+      </DetailValue>
+    </DetailsList>
+  </WidePopover>
+);
+
+const SchedulingInfo = ({
+  executeEveryMs,
+  searchWithinMs,
+  scheduler,
+  title,
+  clearNotifications,
+}:{
+  executeEveryMs: number,
+  searchWithinMs: number,
+  scheduler: Scheduler, title: string,
+  clearNotifications: () => void
+}) => {
+  const executeEveryFormatted = moment.duration(executeEveryMs)
+    .format('d [days] h [hours] m [minutes] s [seconds]', { trim: 'all', usePlural: false });
+  const searchWithinFormatted = moment.duration(searchWithinMs)
+    .format('d [days] h [hours] m [minutes] s [seconds]', { trim: 'all' });
+
+  return (
+    <>
+      {`Runs every ${executeEveryFormatted}, searching within the last ${searchWithinFormatted}. `}
```

Comment:
Can we remove the trailing whitespace?
```suggestion
      {`Runs every ${executeEveryFormatted}, searching within the last ${searchWithinFormatted}.`}
```

Reply:
We need that space to separate the texts and the button.